src/main/java/teammates/ui/pagedata/AdminActivityLogPageData.java

```diff
@@ -171,76 +183,54 @@ private boolean shouldExcludeLogEntry(ActivityLogEntry logEntry) {
         if (q == null) {
             if (this.queryMessage == null) {
                 this.queryMessage = "Error parsing the query. QueryParameters not created.";
             }
-            logEntry.setToShow(true);
-            return logEntry;
+            return true;
         }
         
         //Filter based on what is in the query
         if (q.isRequestInQuery && !arrayContains(q.requestValues, logEntry.getServletName())) {
-            logEntry.setToShow(false);
-            return logEntry;
+            return false;
         }
         if (q.isResponseInQuery && !arrayContains(q.responseValues, logEntry.getAction())) {
-            logEntry.setToShow(false);
-            return logEntry;
+            return false;
         }
         if (q.isPersonInQuery
                 && !logEntry.getName().toLowerCase().contains(q.personValue.toLowerCase())
                 && !logEntry.getGoogleId().toLowerCase().contains(q.personValue.toLowerCase())
                 && !logEntry.getEmail().toLowerCase().contains(q.personValue.toLowerCase())) {
-            logEntry.setToShow(false);
-            return logEntry;
+            return false;
         }
         if (q.isRoleInQuery && !arrayContains(q.roleValues, logEntry.getRole())) {
-            logEntry.setToShow(false);
-            return logEntry;
+            return false;
         }
         if (q.isCutoffInQuery) {
-            if (logEntry.getTimeTaken() == null) {
-                logEntry.setToShow(false);
-                return logEntry;
+            if (logEntry.getTimeTaken() == 0) {
+                return false;
             }
             
             if (logEntry.getTimeTaken() < q.cutoffValue) {
-                logEntry.setToShow(false);
-                return logEntry;
+                return false;
             }
         }
         if (q.isInfoInQuery) {
             
             for (String keyString : q.infoValues) {
-                if (!logEntry.getMessageInfo().toLowerCase().contains(keyString.toLowerCase())) {
-                    logEntry.setToShow(false);
-                    return logEntry;
+                if (!logEntry.getMessage().toLowerCase().contains(keyString.toLowerCase())) {
+                    return false;
                 }
             }
             
-            logEntry.setToShow(true);
-            logEntry.setKeyStringsToHighlight(q.infoValues);
-            logEntry.highlightKeyStringInMessageInfoHtml();
-            return logEntry;
+            return true;
         }
         if (q.isIdInQuery && !arrayContains(q.idValues, logEntry.getId())) {
-            logEntry.setToShow(false);
-            return logEntry;
-        }
-        
-        if (shouldExcludeLogEntry(logEntry)) {
-            logEntry.setToShow(false);
-            return logEntry;
+            return false;
         }
         
-        logEntry.setToShow(true);
-        return logEntry;
+        return !shouldExcludeLogEntry(logEntry);
```

Comment:
If this is the only usage of the method `shouldExcludeLogEntry`, see if you can make it to `shouldIncludeLogEntry` instead.